### PR TITLE
Enable CA1822 in Visual Studio layer

### PR DESCRIPTION
**Pull request by Youssef1313**
The issues stated were fixed and closed.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -271,6 +271,6 @@ csharp_style_var_elsewhere = true:warning
 
 [src/{VisualStudio}/**/*.{cs,vb}]
 # CA1822: Make member static
-# Not enforced as a build 'warning' for 'VisualStudio' layer due to large number of false positives from https://github.com/dotnet/roslyn-analyzers/issues/3857 and https://github.com/dotnet/roslyn-analyzers/issues/3858
-# Additionally, there is a risk of accidentally breaking an internal API that partners rely on though IVT.
-dotnet_diagnostic.CA1822.severity = suggestion
+# There is a risk of accidentally breaking an internal API that partners rely on though IVT.
+dotnet_diagnostic.CA1822.severity = warning
```
Comment (Youssef1313) [inline .editorconfig:275]: This looks redundant as it's already specified above, but it's required due to #50577

File: src/VisualStudio/CSharp/Test/Interactive/Commands/ResetInteractiveTests.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Interactive.Commands
     [UseExportProvider]
     public class ResetInteractiveTests
     {
-        private string WorkspaceXmlStr =>
+        private const string WorkspaceXmlStr =
```
Comment (Youssef1313) [inline src/VisualStudio/CSharp/Test/Interactive/Commands/ResetInteractiveTests.cs:29]: This is the only change to `const`.

**Comment by mavasani** [comment]
@Youssef1313 FYI that I already attempted this locally in the past, but there are lot of internal APIs in the VisualStudio layer that are used via IVTs by internal partner teams, so this poses a big risk of introducing breaking changes. I think this will be a risky change and would certainly require validation and approval from @sharwell and @jinujoseph

**Comment by Youssef1313** [comment]
> @Youssef1313 FYI that I already attempted this locally in the past, but there are lot of internal APIs in the VisualStudio layer that are used via IVTs by internal partner teams, so this poses a big risk of introducing breaking changes. I think this will be a risky change and would certainly require validation and approval from @sharwell and @jinujoseph

@mavasani I'm only changing private APIs. Any internal/public APIs are not touched.

**Comment by jinujoseph** [comment]
Thanks for the contribution @Youssef1313, Blocking this temporarily as we discuss with the team.

**Comment by Youssef1313** [comment]
@jinujoseph @mavasani Any updates on this? Note that I'm only touching `private` APIs, which I think aren't exposed via `InternalsVisibleTo`, so there shouldn't be a problem.

**Comment by mavasani** [comment]
@Youssef1313 thanks, given it is only for private APIs, it should be fine. I’ll review the PR. Thanks for your patience.

**Review by sharwell — APPROVED** [review]

**Comment by Youssef1313** [comment]
@mavasani This is passing CI now.

**Review by mavasani — APPROVED** [review]

**Pull request merged**